src/Microsoft.Windows.ComputeVirtualization/Process.cs

```diff
@@ -117,12 +115,17 @@ public bool WaitForExit(int timeout)
         /// </remarks>
         public void Kill()
         {
-            if (!_killed)
+            KillAsync().Wait();
+        }
+
+        public async Task KillAsync()
+        {
+            string result;
+            if (!_killed && !_exited && HcsFunctions.ProcessHcsCall(HcsFunctions.HcsTerminateProcess(_p, out result), result))
             {
-                // Ignore the error code.
-                HcsFunctions.TerminateProcessInComputeSystem(_id, _pid);
-                _killed = true;
+                await _watcher.WatchAsync(HCS_NOTIFICATIONS.HcsNotificationProcessExited);
             }
+            _killed = true;
```

Comment:
We chatted.  There is some difference in meaning here, especially when you consider .ExitCode.